### ISSUE TITLE
AttributeError: 'ImageModelDescriptor' object has no attribute 'cpu'

### ISSUE THREAD

**Issue by WangGosh** [Suzie1/ComfyUI_Comfyroll_CustomNodes#172]
I deployed ComfyUI on the DSW server of Alibaba Cloud. When I tried to zoom in with the enlarged model, it displayed this error
![image](https://github.com/Suzie1/ComfyUI_Comfyroll_CustomNodes/assets/163592591/7cc3a1d0-f1b1-4523-96c2-7fe7e7610d17)
![image](https://github.com/Suzie1/ComfyUI_Comfyroll_CustomNodes/assets/163592591/6ab8fb71-5e51-447a-91e7-6f8a5225be0e)
![image](https://github.com/Suzie1/ComfyUI_Comfyroll_CustomNodes/assets/163592591/bf1e1e1a-fc50-44b6-a5a8-b166ff314502)

**Comment by WangGosh** [comment]
pytorch2.1.2tensorflow2.14.0-gpu-py310-cu121-ubuntu22.04
This is my system environment

**Comment by Moxie1776** [comment]
I stopped using a1111 for a few months. When I tried it today again today, I'm getting the same error...

**Comment by speters** [comment]
Can confirm this when trying out the [DemonCore Workflow from Civitai](https://civitai.com/articles/5862/workflow-for-demoncore-41), which depends on these nodes for upscaling.

After having searched and found a [hint on stackoverflow](https://stackoverflow.com/questions/64897730/how-to-run-a-pytorch-project-with-cpu), I made a desperate attempt that finally worked:

```patch
diff --git a/nodes/functions_upscale.py b/nodes/functions_upscale.py
index 2c25f03..a6c748c 100644
--- a/nodes/functions_upscale.py
+++ b/nodes/functions_upscale.py
@@ -54,7 +54,8 @@ def upscale_with_model(upscale_model, image):
             if tile < 128:
                 raise e

-    upscale_model.cpu()
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    upscale_model.to(device)
     s = torch.clamp(s.movedim(-3,-1), min=0, max=1.0)
     return s

```

**Comment by DGdev91** [comment]
> Can confirm this when trying out the [DemonCore Workflow from Civitai](https://civitai.com/articles/5862/workflow-for-demoncore-41), which depends on these nodes for upscaling.
> 
> After having searched and found a [hint on stackoverflow](https://stackoverflow.com/questions/64897730/how-to-run-a-pytorch-project-with-cpu), I made a desperate attempt that finally worked:
> 
> ```diff
> diff --git a/nodes/functions_upscale.py b/nodes/functions_upscale.py
> index 2c25f03..a6c748c 100644
> --- a/nodes/functions_upscale.py
> +++ b/nodes/functions_upscale.py
> @@ -54,7 +54,8 @@ def upscale_with_model(upscale_model, image):
>              if tile < 128:
>                  raise e
> 
> -    upscale_model.cpu()
> +    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
> +    upscale_model.to(device)
>      s = torch.clamp(s.movedim(-3,-1), min=0, max=1.0)
>      return s
> ```

Nice, works great for me. have you already made a PR for this fix?

**Comment by speters** [comment]
No, I didn't make a PR. The @Suzie1 account seems pretty inactive since Feb'24, so I didn't bother.
